bundle/src/main/java/com/adobe/acs/samples/replication/impl/SampleReplicatorImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.adobe.acs.samples.replication.impl;
+
+import com.day.cq.replication.*;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.resource.ResourceResolver;
+
+import javax.jcr.Session;
+
+@Component
+@Service
+public class SampleReplicatorImpl implements SampleReplicator {
+
+    // Replicator OSGi service is needed to add dispatcher flush requests to the replication queue
+    @Reference
+    protected Replicator replicator;
+
+    @Override
+    public void replicate(ResourceResolver resourceResolver, String path, final String agentId,
+                          ReplicationActionType replicationAction) throws ReplicationException {
+
+        final ReplicationOptions options = new ReplicationOptions();
+
+        // AgentFilter selects the replication agents that accept the replication request
+        options.setFilter(new AgentFilter() {
```

Comment:
Good to node and link to OOTB AgentFilters so ppl dont recrease the common use case...

https://docs.adobe.com/docs/en/aem/6-0/develop/ref/javadoc/com/day/cq/replication/AgentFilter.html